### PR TITLE
Feature/issue 1310

### PR DESCRIPTION
**Pull request by sashapanasiuk5**
dev
## Issue

* [Ticket](https://github.com/ita-social-projects/StreetCode/issues/1310)

## Code reviewers

- [ ] @ulyasaur
- [ ] @DmytroHoliaka
- [ ] @darkravchuk

### Second Level Review

- [ ] @github_username

## Summary of issue

The streetcode is created using invalid indexes

## Summary of change

Add rule for index validation 

## Testing approach

Add test for boundary values

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by ulyasaur — APPROVED** [review]

**Review by Spoilsport0 — APPROVED** [review]

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/CreateStreetcodeTests.cs [hunk 1]
```diff
@@ -178,5 +179,35 @@ public async Task AddImagesAsync_ValidImagesIds_DoesNotThrowException()
...
+            // Arrange
+            StreetcodeCreateDTO streetcode = new StreetcodeCreateDTO() { Index = index };
+            var mapperMock = new Mock<IMapper>();
+            var repositoryWrapperMock = new Mock<IRepositoryWrapper>();
+            repositoryWrapperMock.Setup(repo => repo.StreetcodeImageRepository.CreateRangeAsync(It.IsAny<IEnumerable<StreetcodeImage>>()))
+                .Returns(Task.CompletedTask);
+            var loggerServiceMock = new Mock<ILoggerService>();
+            var stringLocalizerAnErrorOccurredMock = new Mock<IStringLocalizer<AnErrorOccurredSharedResource>>();
+            var stringLocalizerFailedToCreateMock = new Mock<IStringLocalizer<FailedToCreateSharedResource>>();
+
```
Comment (KateYatsiuk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/CreateStreetcodeTests.cs:197]: This chunk of code is very hard to read.
Pay attention to the whole file - the same mocks are used by many tests. You can try to consolidate common setup code in a shared setup method.
See some better examples in unit tests for other functional units (e.g. News)

**Review by KateYatsiuk — COMMENTED** [review]
Please consider comments and pay attention to Sonar Cloud analysis (btw, it's nice that you remembered to test your changes, 100% coverage on new code is something you should always strive for when changing existing code base).

P.S. You can delete CreateStreetcodeHandlerTests.cs file (its content is commented and very old) and as a result you'll be able to rename CreateStreetcodeTests.cs to CreateStreetcodeHandlerTests.cs so that it matches naming pattern.

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs [hunk 2]
```diff
@@ -59,6 +59,12 @@ public async Task<Result<int>> Handle(CreateStreetcodeCommand request, Cancellat
             {
                 var streetcode = StreetcodeFactory.CreateStreetcode(request.Streetcode.StreetcodeType);
                 _mapper.Map(request.Streetcode, streetcode);
+                if (streetcode.Index < 1 || streetcode.Index > 9999)
```
[the remark this answers is not in the document]
Comment (sashapanasiuk5) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:62]: Thank you for your feedback. What is the best place to place constants( that replace magic number )? Handler, Entity or something else?

[the remark this answers is not in the document]
Comment (KateYatsiuk) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:62]: There's no official rule for this, because it depends on the context. If it is a number that simply describes a certain condition when an action can be executed or not (especially if the given number is reused in many methods/tests of the class), then they are usually declared in that specific class to ensure visibility to other methods. If it directly concerns the validation of some field, then it should be placed in the validator (e.g. FluentValidations validator). _Read about validation at the end of the comment._ When there are much more of these constants and they apply to the same context, a static class may be created where they are grouped, then these constants are not spread over all files.

Taking into account all the circumstances of this particular project (which also concerns validation (or its absence)), you can place them in the Handler for now.

In my opinion, the backend of this project suffers from insufficient validation or its absence in many places (not only DB entities). I opened [a ticket](https://github.com/ita-social-projects/StreetCode/issues/308) for this a long time ago, but it seems that this task has not been completed yet.

**Review by DmytroHoliaka — APPROVED** [review]
